modules/controller/lib/request_handler.js

```diff
@@ -0,0 +1,26 @@
+const verifyContentType = require('./verify_content_type');
+
+module.exports = function (opts) {
+  var controller = opts.controller;
+  var payload = opts.payload;
+  var responder = opts.responder;
+  var method = opts.method;
+
+  return function (request, response, next) {
+    var err;
+    var headers = request.headers;
+    if (method === 'update' || method === 'create') {
```

Comment:
Since this handler is shared in all requests, I was trying to avoid doing any method checks inside here (hence the validateRequest abstraction), but maybe that was premature optimization.